File: android_java/demo_list/app/src/main/java/com/example/demo_list/MainActivity.java
package com.example.demo_list;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    private List<Student> dataList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        dataList = new ArrayList<>();
        addStudent();

    }

    @Override
    protected void onStart() {
        super.onStart();
        ListView listView = findViewById(R.id.list_view);
        ListAdapter adapter = new ListAdapter(this, dataList, R.layout.layout_list_item);
        listView.setAdapter(adapter);
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                String desc = dataList.get(position).getDesc();
                Toast.makeText(MainActivity.this, desc, Toast.LENGTH_SHORT).show();
            }
        });
    }

    private void addStudent() {
        for (int i = 1; i < 30; i++) {
            Student student = new Student();
            student.setName("Tom(" + i + ")");
            student.setDesc("Tom(" + i + ")" + " is a good boy!");
            dataList.add(student);
        }
    }

}

File: android_kotlin/demo_list/app/src/main/java/com/example/demo_list/MainActivity.kt
package com.example.demo_list

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), AdapterView.OnItemClickListener {
    private val dataList: MutableList<Student> = ArrayList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        addStudent()
    }

    override fun onStart() {
        super.onStart()
        val adapter = ListAdapter(this, dataList, R.layout.layout_list_item)
        list_view.adapter = adapter
        list_view.onItemClickListener = this
//        list_view.setOnItemClickListener { _, _, position, _ ->
//            val student = dataList[position]
//            Toast.makeText(this, student.desc, Toast.LENGTH_SHORT).show()
//        }
    }

    private fun addStudent() {
        for (i in 1..30) {//.. 双闭，until 前闭后开；downTo 降序
            val student = Student("Tom($i)", "Tom($i) is a good boy!")
            dataList.add(student)
        }
    }

    override fun onItemClick(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
        val student = dataList[position]
        Toast.makeText(this, student.desc, Toast.LENGTH_SHORT).show()
    }
}
